packages/compat/src/fixup-rules.js

```diff
@@ -98,31 +138,105 @@ export function fixupRule(ruleDefinition) {
 		: ruleDefinition.create.bind(ruleDefinition);
 
 	function ruleCreate(context) {
+		const sourceCode = context.sourceCode;
+		const needsContextShims = !("getCwd" in context);
```

Comment:
Since we're patching `context` regardless of this (for ESLint v9), I think a name like `isESLint10` would be more descriptive.